### DISCUSSION TITLE
No '@sap/cds-gen' package found.

### DISCUSSION THREAD

**Discussion by ThePlenkov** [cap-js/.github#6] [General]
Hi!

I'd like to try a new command available in the cds CLI. However currently i have this:
```
 cds gen 'Hotel Booking app'

No '@sap/cds-gen' package found. Install it with

  npm i -g @sap/cds-gen
```

the problem is - when I try to install this package a i have this:
```
 npm i -g @sap/cds-gen
npm error code E404
npm error 404 Not Found - GET https://registry.npmjs.org/@sap%2fcds-gen - Not found
npm error 404
npm error 404  '@sap/cds-gen@*' is not in this registry.
npm error 404
npm error 404 Note that you can also install from a
npm error 404 tarball, folder, http url, or git url.

npm error A complete log of this run can be found in: /home/codespace/.npm/_logs/2024-08-16T11_31_57_936Z-debug-0.log
```
Are you using some alternative npm registry for this package?

Thank you!

**Comment by ThePlenkov** [comment]
@schwma do you may be know? Thank you!

**Comment by David-Kunz** [comment]
Hi @ThePlenkov ,

The functionality of `@sap/cds-gen` is only available in [SAP Build Code](https://www.sap.com/products/technology-platform/developer-tools.html) as part of the Joule capabilities and is not externally available on npm.

Best regards,
David
